File: nodejs/6 kyu/162.js
// https://www.codewars.com/kata/545f05676b42a0a195000d95/train/javascript

// Passed

function ranks(a) {
  let sortedArr = [...a].sort((a, b) => b - a); // sorts in descending order
  let numRankObject = new Object();
  let rank = 1;
  for (const num of sortedArr) {
    if (!(num in numRankObject)) {
      numRankObject[num] = rank;
    }
    rank++;
  }

  let rankArr = a.map((num) => numRankObject[num]);
  return rankArr;
}

const output = ranks([5, 2, 3, 5, 5, 4, 9, 8, 0]);
console.log(output);

File: python/7 kyu/162.py
# https://www.codewars.com/kata/5a63948acadebff56f000018/train/python

# Passed

def max_product(lst, n_largest_elements):
    lst.sort(reverse=True)
    result = 1
    for idx in range(n_largest_elements):
        result *= lst[idx]
        
    return result

output = max_product([10,2,3,8,1,10,4], 5)
print(output)